src/battlearena/SearchPlayers.js

```diff
@@ -30,11 +31,39 @@ class SearchPlayers extends React.Component<Props, State> {
 		this.handleKeyPress.bind(this);
 
 		this.state = {
-			playerList: ["Jim", "John", "Jimfrey", "Eich", "Eichers", "Baylor", "Jorge", "David", "Emil", "Adam"], // NEEDS API CALL HERE
+			playerList: [], // NEEDS API CALL HERE
 			searchList: []
 		}
 	}
 
+	componentDidMount(): void {
+
+		// Set the cookie and update state.
+		const cookies = new Cookies();
+		const token = cookies.get('token');
+		const responsePromise: Promise<Response> = fetch('/api/user/allUsers', {
+			method: 'GET',
+			headers: {
+				'Access-Control-Allow-Origin': '*',
+				'Content-Type': 'application/json',
+				'Access-Control-Allow-Credentials': 'true',
+				'x-auth-token': token
+			},
+		});
+		responsePromise.then (
+			response => response.json().then(data => {
+				if (response.status !== 200) {
+					console.log(response.status);
+					console.log(data.msg);
+				}
+				else {
+					console.log(data.map(usernames => usernames.userName));

```

Comment:
nice, that's clean

Reply:
As for what you need to put in the user class, pretty much all of the stuff that gets returned here (that you're currently just throwing away) should probably get put in there. 

You probably know this already, but just in case you don't, you can really easily see all the things an object has by doing console.log() and then looking through it in your JS console on your browser